packages/web/components/Dashboard/Settings/Subscription/SubscriptionForm.tsx

```diff
@@ -0,0 +1,165 @@
+  })
+
+  const handleCancelSubscription = async () => {
+    if (!(await confirmCancellation())) return
+    
+    updateSubscriptionRenewal({
+      variables: {
+        cancelAtPeriodEnd: true,
+      }
+    })
+  }
+
+  const SubscriptionStatusBadge = () => {
+    return (
+      <>
+        <span className="badge">
+          {user.isPremiumUser ? 'premium user' : 'free user'}
+        </span>
+        <style jsx>{`
+          .badge {
+            padding: 10px;
+            border-radius: 5px;
+            text-transform: uppercase;
+            font-size: ${theme.typography.paragraphSM}
+            font-weight: 600;
+            background-color: ${!user.isPremiumUser ? theme.colors.gray100 : theme.colors.greenLight};
+            color: ${!user.isPremiumUser ? theme.colors.gray600 : theme.colors.greenDark};
+          }
+        `}</style>
+      </>
+    )
+  }
+
+  return (
+    <>
+      {showPaymentFormModal && (
+        <PaymentFormModal onClose={() => setShowPaymentFormModal(false)} />
+      )}
+      <div className="page-container">
+        <p style={{ marginBottom: '20px' }}>{t('subscription.copy')}</p>
+        <p style={{ marginBottom: '10px' }}><strong>Subscription status:</strong> <SubscriptionStatusBadge /></p>
+        {user.isPremiumUser && (
+          <>
+            <p><strong>Current Plan:</strong> {subscriptionPlan}</p>
+            {user.membershipSubscription?.lastFourCardNumbers && (
+              <CardOnFile last4="1234" onUpdateCard={() => {}} />
+            )}
+            {isCancelling ? (
+              <>
+                <p>Your subscription will end on <strong style={{ color: theme.colors.red }}>{formatLongDate(user?.membershipSubscription?.expiresAt)}</strong></p>
+                {/* Add mutation for this */}
+                <Button
+                  variant={ButtonVariant.Link}
+                  onClick={() => {
+                  updateSubscriptionRenewal({
+                    variables: {
+                      cancelAtPeriodEnd: false,
+                    }
+                  })
+              }}>Reactivate subscription</Button>
+              </>
+            ) : (
+              <>
+                <p style={{ marginBottom: '20px' }}><strong>Next billing date:</strong> Your subscription will renew on <strong>{formatLongDate(user?.membershipSubscription?.expiresAt)}</strong></p>
+              </>
+            )}
+          </>
+        )}
+        {showPaymentForm && (
+          <PaymentForm />
+        )}
+        {!user?.membershipSubscription?.cancelAtPeriodEnd && !showPaymentForm && (
+          <>
+            <Button
+              onClick={() => {
+                setShowPaymentForm(true)
+              }}
+              variant={ButtonVariant.Link}
+            >
+              Change Plan
+            </Button>
+            <Button
+              onClick={handleCancelSubscription}
+              variant={ButtonVariant.Link}
+              style={{
```

Comment:
inline style

Reply:
🙅🏻‍♂️ Gone!

Reply:
I wasn't able to avoid this one so far because even with the global selector with parent class/descendent selector we used the other day, it ends up impacting the other buttons since we have several in the component